src/client/components/pages/parts/pager.js

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (C) 2019 Prabal Singh
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+import * as bootstrap from 'react-bootstrap';
+import PropTypes from 'prop-types';
+import React from 'react';
+import request from 'superagent-bluebird-promise';
+
+
+const {Pager, Button, ButtonGroup, DropdownButton, MenuItem} = bootstrap;
+
+class PagerElement extends React.Component {
+	constructor(props) {
+		super(props);
+		this.state = {
+			from: this.props.from,
+			nextEnabled: this.props.results.length === this.props.size,
+			query: this.props.query,
+			results: this.props.results,
+			size: this.props.size
+		};
+		this.handleClickPrevious = this.handleClickPrevious.bind(this);
+		this.handleClickNext = this.handleClickNext.bind(this);
+		this.handleResultsPerPageChange = this.handleResultsPerPageChange.bind(this);
+		this.triggerSearch = this.triggerSearch.bind(this);
+	}
+
+	componentDidUpdate(prevProps, prevState) {
+		if (prevProps.query !== this.props.query) {
+			// eslint-disable-next-line react/no-did-update-set-state
+			this.setState(this.props, this.triggerSearch);
```

Comment:
I guess you should only set the query prop here (`{query: this.props.query}`), even though React is optimized and should not change the other properties if they are the same value.

It'll make the code clearer to understand though, and I think it might allow you to remove the `react/no-did-update-set-state` comment. (I haven't checked, don't quote me on that :p )

Reply:
Yeah alright. I'll have to update `from : 0` then.
Also should the `size` be updated? If I'm searching something and I've set size to 100, I'd want the new results table with size 100 only right?

Reply:
For `from` I think you don't need to react to any update, considering only the PagerElement changes `from`.
We do want to pass it from the parent element, but only so that we can for example add `?from=150`  to the url and have that be passed down to the pager.

Unless I'm mistaken, the same thing applies for the `size`: you only modify it in PagerElement, so there's no need to react to props changes.

Reply:
I'll have a look later